ddp-workspace/projects/toolkit/src/lib/components/app/app-redesigned-base.component.ts

```diff
@@ -0,0 +1,62 @@
+import { OnInit, OnDestroy, Inject } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { NoopScrollStrategy } from '@angular/cdk/overlay';
+import { NavigationEnd, Router } from '@angular/router';
+import { CommunicationService } from '../../services/communication.service';
+import { ToolkitConfigurationService } from '../../services/toolkitConfiguration.service';
+import { JoinMailingListComponent } from '../dialogs/joinMailingList.component';
+import { SessionWillExpireComponent } from '../dialogs/sessionWillExpire.component';
+import { CompositeDisposable, RenewSessionNotifier } from 'ddp-sdk';
+
+export class AppRedesignedBaseComponent implements OnInit, OnDestroy {
+    private anchor = new CompositeDisposable();
+    private readonly DIALOG_BASE_SETTINGS = {
+        width: '740px',
+        position: { top: '30px' },
+        data: {},
+        autoFocus: false,
+        scrollStrategy: new NoopScrollStrategy()
+    };
+
+    constructor(
+        private communicationService: CommunicationService,
+        private dialog: MatDialog,
+        private renewNotifier: RenewSessionNotifier,
+        private router: Router,
+        @Inject('toolkit.toolkitConfig') private config: ToolkitConfigurationService) { }
+
+    public ngOnInit(): void {
+        this.initMailingListDialogListener();
+        this.initSessionExpiredDialogListener();
+        this.initRouterListener();
+    }
+
+    public ngOnDestroy(): void {
+        this.anchor.removeAll();
+    }
+
+    private initMailingListDialogListener(): void {
+        const modalOpen = this.communicationService.openJoinDialog$.subscribe(() => {
+            this.dialog.open(JoinMailingListComponent, this.DIALOG_BASE_SETTINGS);
```

Comment:
Does the app*component have to be the one that handles the dialog? Already A LOT going on here. Perhaps this can be setup in a separate component?
Also, we should try to avoid an explicit reference to JoinMailingListComponent from this component; this component is meant to be generic/high-level/reusable and JoinMailingListComponent is specific/different level of abstraction.
Maybe the dialog class could be injected into a different component?

Reply:
Hm, AppRedesignedBaseComponent is only 63 lines and 3 methods, I don't think that there is going on a lot :) 
It's a bit unclear for me why we should avoid JoinMailingListComponent here: AppRedesignedBaseComponent will be a universal component for all redesigned CMI studies, and at least 2 studies already use this component - osteo and brain-v2. And all CMI studies use JoinMailingListComponent.

Reply:
But maybe we can do this base component more universal - we can make constructor parameters optional using @Optional() decorator, so then it allows disabling initMailingListDialogListener or initSessionExpiredDialogListener if we don't need some of them.

Reply:
Actually, I guess before I suggest anything, I am not sure if I understand the purpose of this component. You also have an app.component.ts that appears to wire all the the functionality of the app together, but "redesigned" is not connect to it.
So is this meant to be the updated base component for the newer style apps (e.g., Osteo) and we would use app.component.ts just for the older apps (e.g., Angio). I am getting this right?

Reply:
Yes, angio, brain, mbc, mpc use app.component.ts
osteo, brain-v2 use redesigned as a base component, which inits global listeners

Reply:
OK. No problem. Let's move forward with it.